src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -971,10 +970,11 @@ public static unsafe IntPtr StringToHGlobalAnsi(string s)
                 return IntPtr.Zero;
             }
 
-            int nb = (s.Length + 1) * SystemMaxDBCSCharSize;
```

Comment:
I personally would prefer to compute the maximum possible string length as a constant and then check that `Length` is less than that. It would remove this unnecessary and easily done incorrectly math to be done in a single place.